mayastor/src/bdev/nexus/nexus_child.rs

```diff
@@ -403,6 +423,115 @@ impl NexusChild {
+            .dma_malloc((blocks * block_size) as usize)
+            .context(MetaDataHeaderAlloc {})?;
+        self.read_at((partition_lba + 1) * block_size, &mut buf)
+            .await
+            .context(MetaDataHeaderRead {})?;
+        let header = NexusMetaData::read_header(&buf)
+            .context(MetaDataHeaderInvalid {})?;
+
+        //
+        // Index
+        let index = if header.used_entries > 0 {
+            let blocks = Unsigned::get_aligned_blocks(
+                (header.used_entries * header.entry_size) as u64,
+                block_size,
+            );
+            let mut buf = desc
+                .dma_malloc((blocks * block_size) as usize)
+                .context(MetaDataIndexAlloc {})?;
+            self.read_at(
+                (header.self_lba + header.index_start) * block_size,
+                &mut buf,
+            )
+            .await
+            .context(MetaDataIndexRead {})?;
+            NexusMetaData::read_index(&buf, &header)
+                .context(MetaDataIndexInvalid {})?
+        } else {
+            NexusMetaData::empty_index(&header)
+                .context(MetaDataIndexInvalid {})?
+        };
+
+        Ok(NexusMetaData {
+            header,
+            index,
+        })
+    }
+
+    pub async fn probe_config_object(
+        &self,
+        metadata: &NexusMetaData,
+        n: u32,
+    ) -> Result<NexusConfig, ChildError> {
+        if n >= metadata.header.used_entries {
+            return Err(ChildError::LabelInvalid {});
+        }
+
+        let entry = &metadata.index[n as usize];
+
+        let (bdev, desc) = self.get_dev()?;
+        let block_size = bdev.block_len() as u64;
+
+        let blocks = entry.data_end - entry.data_start + 1;
+        let mut buf = desc
+            .dma_malloc((blocks * block_size) as usize)
+            .context(MetaDataObjectAlloc {})?;
+        self.read_at(
+            (metadata.header.self_lba + entry.data_start) * block_size,
+            &mut buf,
+        )
+        .await
+        .context(MetaDataObjectRead {})?;
+
+        Ok(NexusMetaData::read_config_object(&buf, entry)
+            .context(MetaDataObjectInvalid {})?)
+    }
+
+    pub async fn probe_all_config_objects(
+        &self,
+        metadata: &NexusMetaData,
+    ) -> Result<Vec<NexusConfig>, ChildError> {
+        let (bdev, desc) = self.get_dev()?;
+        let block_size = bdev.block_len() as u64;
+
+        let mut list: Vec<NexusConfig> = Vec::new();
+
+        for entry in &metadata.index {
+            let blocks = entry.data_end - entry.data_start + 1;
+            let mut buf = desc
+                .dma_malloc((blocks * block_size) as usize)
+                .context(MetaDataObjectAlloc {})?;
+            self.read_at(
+                (metadata.header.self_lba + entry.data_start) * block_size,
+                &mut buf,
+            )
+            .await
+            .context(MetaDataObjectRead {})?;
+            list.push(
```

Comment:
what happens this allocated memory? This memory is precious, and if you need to keep it around, we should copy it out, and free the DMA-able memory, makes sense?